### PR TITLE
Some cleaning [server]

### PR DESCRIPTION
**Pull request by pblottiere**
## Description

- Remove unused functions
- Rename some getter
- Use more enum instead of string
- Add unit test

### CODE REVIEW

File: src/server/services/wms/qgswmsgetprint.cpp [hunk 1]
```diff
@@ -25,51 +25,36 @@
...
-    QgsServerRequest::Parameters params = request.parameters();
-
-    Q_UNUSED( version );
-
-    QgsWmsParameters wmsParameters( QUrlQuery( request.url() ) );
+    const QgsWmsParameters wmsParameters( QUrlQuery( request.url() ) );
     QgsRenderer renderer( serverIface, project, wmsParameters );
 
-    QString format = params.value( "FORMAT" );
+    const QgsWmsParameters::Format format = wmsParameters.format();
```
Comment (nyalldawson) [inline src/server/services/wms/qgswmsgetprint.cpp]: Good news -- it looks like soon we'll be able to run a clang-tidy fixit over the whole code and automatically add all these missing consts: https://reviews.llvm.org/D45444

When we've done that let's crack down on this and make it standard QGIS coding practice to constify anything which can be.

File: src/server/services/wms/qgswmsparameters.h [hunk 2]
```diff
@@ -503,6 +506,13 @@ namespace QgsWms
        */
       QString formatAsString() const;
 
+      /**
+       * Returns format parameter as a string.
+       * \returns format parameter as a string.
```
Comment (nyalldawson) [inline src/server/services/wms/qgswmsparameters.h]: No need for redundant \returns here.

File: src/server/services/wms/qgswmsparameters.cpp [hunk 3]
```diff
@@ -716,18 +718,39 @@ namespace QgsWms
     return mWmsParameters[ QgsWmsParameter::FORMAT ].toString();
   }
 
-  QgsWmsParameters::Format QgsWmsParameters::format() const
+  QString QgsWmsParameters::formatAsString( const QgsWmsParameters::Format format )
```
Comment (m-kuhn) [inline src/server/services/wms/qgswmsparameters.cpp]: const method?

Comment (pblottiere) [reply src/server/services/wms/qgswmsparameters.cpp]: Actually it's a static method.

File: src/server/services/wms/qgswmsgetprint.cpp (hunk 1 above)
Comment (pblottiere) [reply src/server/services/wms/qgswmsgetprint.cpp]: > When we've done that let's crack down on this and make it standard QGIS coding practice to constify anything which can be.

Yesss!!!!!

File: src/server/services/wms/qgswmsparameters.cpp (hunk 3 above)
Comment (m-kuhn) [reply src/server/services/wms/qgswmsparameters.cpp]: Ouch :see_no_evil:

File: src/server/services/wms/qgswmsparameters.h (hunk 2 above)
Comment (pblottiere) [reply src/server/services/wms/qgswmsparameters.h]: Done

**Comment by rldhont** [comment]
@pblottiere do you think, you will have time to fix the test ?

**Pull request closed without merging**